lib/pbench/test/unit/server/test_user_management_cli.py

```diff
@@ -0,0 +1,257 @@
+    def test_admin_user_creation(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            result = runner.invoke(
+                user_create,
+                args=[
+                    TestUserManagement.USER_SWITCH,
+                    TestUserManagement.USER_TEXT,
+                    TestUserManagement.PSWD_SWITCH,
+                    TestUserManagement.PSWD_TEXT,
+                    TestUserManagement.EMAIL_SWITCH,
+                    TestUserManagement.EMAIL_TEXT,
+                    TestUserManagement.FIRST_NAME_SWITCH,
+                    TestUserManagement.FIRST_NAME_TEXT,
+                    TestUserManagement.LAST_NAME_SWITCH,
+                    TestUserManagement.LAST_NAME_TEXT,
+                    TestUserManagement.ROLE_SWITCH,
+                    "ADMIN",
+                ],
+            )
+            assert result.exit_code == 0
+            assert result.stdout == "Admin user test_user registered\n"
+            assert not result.stderr_bytes
+
+    @staticmethod
+    @responses.activate
+    def test_user_creation_with_invalid_role(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            result = runner.invoke(
+                user_create,
+                args=[
+                    TestUserManagement.USER_SWITCH,
+                    TestUserManagement.USER_TEXT,
+                    TestUserManagement.PSWD_SWITCH,
+                    TestUserManagement.PSWD_TEXT,
+                    TestUserManagement.EMAIL_SWITCH,
+                    TestUserManagement.EMAIL_TEXT,
+                    TestUserManagement.FIRST_NAME_SWITCH,
+                    TestUserManagement.FIRST_NAME_TEXT,
+                    TestUserManagement.LAST_NAME_SWITCH,
+                    TestUserManagement.LAST_NAME_TEXT,
+                    TestUserManagement.ROLE_SWITCH,
+                    "ADMN",
+                ],
+            )
+            assert result.exit_code == 2
+
+    @staticmethod
+    @responses.activate
+    def test_valid_user_delete(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            TestUserManagement.test_args(server_config, pytestconfig)
+
+            result = runner.invoke(user_delete, args=[TestUserManagement.USER_TEXT])
+            assert result.exit_code == 0
+            assert result.stdout == "User test_user deleted\n"
+            assert not result.stderr_bytes
+
+    @staticmethod
+    @responses.activate
+    def test_invalid_user_delete(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            # Give username that doesn't exists to delete
+            result = runner.invoke(user_delete, args=["wrong_username"])
+            assert result.exit_code == 1
+            assert result.stdout == "User wrong_username does not exists\n"
```

Comment:
Is this an error though? I mean if the username doesn't exist is not necessarily an error of the command, the command did everything right but didn't find the user. Thats why `stdout` instead of `stderr`

Reply:
It's a "user error" to attempt to delete a user that doesn't exist. They should be notified, and especially if we want to grow to supporting filtering on the output we should write "out of band" messages (anything that's not the requested output) to `stderr` so they don't have to account for that in the filter.

Reply:
Stuff that the command is supposed to produce should go to `stdout`.  (I'm not coming up with _anything_ that a typical delete operation would print out.)

Stuff about the __operation__ of the command, particularly descriptions of errors or other "surprising" results, should be sent to `stderr`.  (This would include "verbose mode" confirmations that the deletion occurred, as well as reports when it didn't occur, for whatever reason.)